collector/pg_oldest_blocked.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const oldestBlockedSubsystem = "oldest_blocked"
+
+func init() {
+	registerCollector(oldestBlockedSubsystem, defaultDisabled, NewPGOldestBlockedCollector)
+}
+
+type PGOldestBlockedCollector struct {
+	log log.Logger
+}
+
+func NewPGOldestBlockedCollector(config collectorConfig) (Collector, error) {
+	return &PGOldestBlockedCollector{log: config.logger}, nil
+}
+
+var (
+	oldestBlockedAgeSeconds = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, oldestBlockedSubsystem, "age_seconds"),
+		"Largest number of seconds any transaction is currently waiting on a lock",
+		[]string{},
+		prometheus.Labels{},
+	)
+
+	oldestBlockedQuery = `
+	SELECT
+		coalesce(extract('epoch' from max(clock_timestamp() - state_change)), 0) age_seconds
```

Comment:
Similar to others, we should extract the raw timestamp for this metric.

Reply:
I don't know if that makes sense since we're trying to measure the max duration that a query's been blocked here

Reply:
This can still be done with `time() - metric`.